server/graphql/v2/mutation/AddedFundsMutations.ts

```diff
@@ -0,0 +1,352 @@
+        description: 'The order to edit',
+      },
+      fromAccount: {
+        type: new GraphQLNonNull(GraphQLAccountReferenceInput),
+        description: 'The account that will be used as the source of the funds',
+      },
+      account: {
+        type: new GraphQLNonNull(GraphQLAccountReferenceInput),
+        description: 'The account that will receive the funds',
+      },
+      tier: {
+        type: GraphQLTierReferenceInput,
+        description: 'The tier to which the funds will be added',
+      },
+      amount: {
+        type: new GraphQLNonNull(GraphQLAmountInput),
+        description: 'The total amount of the order, including fees and taxes',
+      },
+      paymentProcessorFee: {
+        type: GraphQLAmountInput,
+        description: 'The payment processor fee amount',
+      },
+      hostFeePercent: {
+        type: GraphQLFloat,
+        description: 'The host fee percent to apply to the order, as a float between 0 and 100',
+      },
+      description: {
+        type: new GraphQLNonNull(GraphQLString),
+        description: 'A short description of the contribution',
+      },
+      memo: {
+        type: GraphQLString,
+        description: 'A private note for the host',
+      },
+      processedAt: {
+        type: GraphQLDateTime,
+        description: 'The date at which the order was processed',
+      },
+      invoiceTemplate: {
+        type: GraphQLString,
+        description: 'The invoice template to use for this order',
+      },
+      tax: {
+        type: GraphQLTaxInput,
+        description: 'The tax to apply to the order',
+      },
+      accountingCategory: {
+        type: GraphQLAccountingCategoryReferenceInput,
+        description: 'The accounting category of this order',
+      },
+    },
+    resolve: async (
+      _,
+      args: Omit<AddFundsMutationArgs, 'transactionsImportRow'> & { order: ObjectReference },
+      req: express.Request,
+    ) => {
+      checkRemoteUserCanUseHost(req);
+
+      const order = await fetchOrderWithReference(args.order, { throwIfMissing: true });
+      const account: Collective = await fetchAccountWithReference(args.account, { throwIfMissing: true });
+      const fromAccount: Collective = await fetchAccountWithReference(args.fromAccount, { throwIfMissing: true });
+      const tier = args.tier && (await fetchTierWithReference(args.tier, { throwIfMissing: true }));
+      const host = await account.getHostCollective({ loaders: req.loaders });
+
+      // Check amounts
+      const totalAmount = getValueInCentsFromAmountInput(args.amount, { expectedCurrency: account.currency });
+      const paymentProcessorFee = args.paymentProcessorFee
+        ? getValueInCentsFromAmountInput(args.paymentProcessorFee, { expectedCurrency: account.currency })
+        : 0;
+
+      validateAddFundsArgs({ account, fromAccount, args, paymentProcessorFee, totalAmount, host, req });
+
+      await twoFactorAuthLib.enforceForAccount(req, host, { onlyAskOnLogin: true });
```

Comment:
Not especially for this PR, but since added funds let host admins change the balance of accounts, we should be stricter than that on 2FA - for both add funds and edit funds.